### PR TITLE
Adopt Psalm

### PR DESCRIPTION
**Pull request by Fantom409**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️/❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  |

### CODE REVIEW

File: src/Provider/FieldProvider.php [hunk 1]
```diff
@@ -17,11 +17,12 @@ public function __construct(array $fieldConfig = [])
...
 
-    /**
-     * @suppress PhanAccessMethodProtected
-     */
+
     public function register(Container $container): void
     {
+        /**
+         * @psalm-suppress InaccessibleMethod
+         */
```
Comment (rustamwin) [inline src/Provider/FieldProvider.php:25]: Let's move it to the method level.

File: src/Provider/FlashProvider.php [hunk 2]
```diff
@@ -12,11 +12,11 @@
...
 final class FlashProvider extends ServiceProvider
 {
-    /**
-     * @suppress PhanAccessMethodProtected
-     */
     public function register(Container $container): void
     {
+        /**
+         * @psalm-suppress InaccessibleMethod
+         */
```
Comment (rustamwin) [inline src/Provider/FlashProvider.php:19]: Let's move it to the method level.

File: src/Provider/I18nProvider.php [hunk 3]
```diff
@@ -24,11 +24,11 @@ public function __construct(string $locale, string $translatePath)
...
     }
 
-    /**
-     * @suppress PhanAccessMethodProtected
-     */
     public function register(Container $container): void
     {
+        /**
+         * @psalm-suppress InaccessibleMethod
+         */
```
Comment (rustamwin) [inline src/Provider/I18nProvider.php:31]: Let's move it to the method level.

File: src/Provider/LoggerProvider.php [hunk 4]
```diff
@@ -19,10 +19,16 @@ final class LoggerProvider extends ServiceProvider
 {
     public function register(Container $container): void
     {
+        /**
+         * @psalm-suppress InaccessibleMethod
+         */
```
Comment (rustamwin) [inline src/Provider/LoggerProvider.php:24]: Let's move it to the method level.

File: src/Provider/LoggerProvider.php [hunk 5]
```diff
@@ -19,10 +19,16 @@ final class LoggerProvider extends ServiceProvider
...
         $container->set(FileRotatorInterface::class, static function () {
             return new FileRotator(10);
         });
 
+        /**
+         * @psalm-suppress InaccessibleMethod
+         */
```
Comment (rustamwin) [inline src/Provider/LoggerProvider.php:31]: Let's move it to the method level.

File: src/Provider/LoggerProvider.php [hunk 6]
```diff
@@ -42,6 +48,9 @@ public function register(Container $container): void
             return $fileTarget;
         });
 
+        /**
+         * @psalm-suppress InaccessibleMethod
+         */
```
Comment (rustamwin) [inline src/Provider/LoggerProvider.php:53]: Let's move it to the method level.

File: src/Provider/MailerInterfaceProvider.php [hunk 7]
```diff
@@ -37,11 +37,12 @@ public function __construct(
...
 
-    /**
-     * @suppress PhanAccessMethodProtected
-     */
+
     public function register(Container $container): void
     {
+        /**
+         * @psalm-suppress InaccessibleMethod
+         */
```
Comment (rustamwin) [inline src/Provider/MailerInterfaceProvider.php:45]: Let's move it to the method level.

File: src/Provider/MiddlewareProvider.php [hunk 8]
```diff
@@ -16,11 +16,11 @@
...
 final class MiddlewareProvider extends ServiceProvider
 {
-    /**
-     * @suppress PhanAccessMethodProtected
-     */
     public function register(Container $container): void
     {
+        /**
+         * @psalm-suppress InaccessibleMethod
+         */
```
Comment (rustamwin) [inline src/Provider/MiddlewareProvider.php:23]: Let's move it to the method level.

**Review by rustamwin — CHANGES_REQUESTED** [review]

**Review by samdark — APPROVED** [review]

**Pull request merged**